internal/exec/exec.go

```diff
@@ -313,16 +313,19 @@ func (r *Request) execList(ctx context.Context, sels []selected.Selection, typ *
 	entryouts := make([]bytes.Buffer, l)
 
 	if selected.HasAsyncSel(sels) {
-		var wg sync.WaitGroup
-		wg.Add(l)
+		concurrency := cap(r.Limiter)
```

Comment:
LGTM. Was worried about Limiter having cap(0) since the API allows you to do that. But it seems everything would just break if someone did that, so should be safe here.